pkg/controller/replication_controller_test.go

```diff
@@ -852,3 +852,105 @@ func TestControllerUpdateStatusWithFailure(t *testing.T) {
 		t.Errorf("Expected 1 get and 2 updates, got %d gets %d updates", gets, updates)
 	}
 }
+
+func doTestControllerBurstReplicas(t *testing.T, burstReplicas, numReplicas int) {
+	client := client.NewOrDie(&client.Config{Host: "", Version: testapi.Version()})
+	fakePodControl := FakePodControl{}
+	manager := NewReplicationManager(client, burstReplicas)
+	manager.podControl = &fakePodControl
+
+	controllerSpec := newReplicationController(numReplicas)
+	manager.controllerStore.Store.Add(controllerSpec)
+
+	expectedPods := 0
+	pods := newPodList(nil, numReplicas, api.PodPending, controllerSpec)
+	var scaleFactor = float64(numReplicas) / float64(burstReplicas)
+
+	// Size up the controller, then size it down, and confirm the expected create/delete pattern
+	for _, numReplicas := range []int{numReplicas, 0} {
+
+		controllerSpec.Spec.Replicas = numReplicas
+		manager.controllerStore.Store.Add(controllerSpec)
+
+		for i := 0.0; i < scaleFactor; i++ {
```

Comment:
Allergic to using comparison operations on floating point numbers :)

(also found the "scale factor" variable name confusing)